tests/integration/relations/opensearch_provider/test_opensearch_provider.py

```diff
@@ -0,0 +1,225 @@
+        await ops_test.model.wait_for_idle(timeout=1200, status="active")
+
+
+@pytest.mark.client_relation
+async def test_database_usage(ops_test: OpsTest):
+    """Check we can update and delete things."""
+    await run_request(
+        ops_test,
+        unit_name=ops_test.model.applications[CLIENT_APP_NAME].units[0].name,
+        relation_id=client_relation.id,
+        method="PUT",
+        endpoint="/albums/_doc/1",
+        payload=re.escape(
+            '{"artist": "Vulfpeck", "genre": ["Funk", "Jazz"], "title": "Thrill of the Arts"}'
+        ),
+    )
+
+    read_index_endpoint = "/albums/_search?q=Jazz"
+    run_read_index = await run_request(
+        ops_test,
+        unit_name=ops_test.model.applications[CLIENT_APP_NAME].units[0].name,
+        endpoint=read_index_endpoint,
+        method="GET",
+        relation_id=client_relation.id,
+    )
+    results = json.loads(run_read_index["results"])
+    logging.info(results)
+    assert results.get("timed_out") is False
+    assert results.get("hits", {}).get("total", {}).get("value") == 1
+    assert (
+        results.get("hits", {}).get("hits", [{}])[0].get("_source", {}).get("artist") == "Vulfpeck"
+    )
+
+
+@pytest.mark.client_relation
+async def test_database_bulk_usage(ops_test: OpsTest):
+    """Check we can update and delete things using bulk api."""
+    bulk_payload = """{ "index" : { "_index": "albums", "_id" : "2" } }
+{"artist": "Herbie Hancock", "genre": ["Jazz"],  "title": "Head Hunters"}
+{ "index" : { "_index": "albums", "_id" : "3" } }
+{"artist": "Lydian Collective", "genre": ["Jazz"],  "title": "Adventure"}
+{ "index" : { "_index": "albums", "_id" : "4" } }
+{"artist": "Liquid Tension Experiment", "genre": ["Prog", "Metal"],  "title": "Liquid Tension Experiment 2"}
+"""
+    await run_request(
+        ops_test,
+        unit_name=ops_test.model.applications[CLIENT_APP_NAME].units[0].name,
+        relation_id=client_relation.id,
+        method="POST",
+        endpoint="/_bulk",
+        payload=re.escape(bulk_payload),
+    )
+
+    read_index_endpoint = "/albums/_search?q=Jazz"
+    run_bulk_read_index = await run_request(
+        ops_test,
+        unit_name=ops_test.model.applications[CLIENT_APP_NAME].units[0].name,
+        endpoint=read_index_endpoint,
+        method="GET",
+        relation_id=client_relation.id,
+    )
+    # TODO assert we're getting the correct value
+    results = json.loads(run_bulk_read_index["results"])
+    logging.info(results)
+    assert results.get("timed_out") is False
+    assert results.get("hits", {}).get("total", {}).get("value") == 3
+    artists = [
+        hit.get("_source", {}).get("artist") for hit in results.get("hits", {}).get("hits", [{}])
+    ]
+    assert set(artists) == {"Herbie Hancock", "Lydian Collective", "Vulfpeck"}
+
+
+@pytest.mark.client_relation
+async def test_database_version(ops_test: OpsTest):
+    """Check version is accurate."""
+    run_version_query = await run_request(
```

Comment:
Would `run_version_response` make more sense here?

Reply:
yes, it would!